distribution/lib/Standard/Table/0.0.0-dev/src/Excel/Excel_Workbook.enso

```diff
@@ -81,7 +96,7 @@ type Excel_Workbook
     ## GROUP Standard.Base.Metadata
        Gets the names of all the sheets.
     sheet_names : Vector Text
-    sheet_names self = Vector.from_polyglot_array (ExcelReader.readSheetNames self.workbook)
```

Comment:
Is there any reason left to use `Vector.from_polyglot_array`? I might have forgotten.

Reply:
Any places that type check for a Vector error if they get an Array which we will need to be careful of.

Reply:
Shouldn't we then keep the conversion? IIRC it's O(1) and this is not even a hot code path since one is not going to query sheet_names too often and there aren't as many of them.

I'd keep the conversion to avoid the issues you mention with Array type.

Reply:
Actually shouldn't we just add
`Vector.from (that:Array) = Vector.from_polyglot_array that`

and then we are done.

Reply:
Hmm, fair point.

However, do the conversions trigger in `case-of` statements? I think that's the place where we 'feel' the Vector-vs-Array difference the most.

And I don't think we _can_ allow conversions there, as then we'd lose a way to check the true type of a value, e.g. if conversions triggered in case-of and we had the conversion you suggested we'd have:
```
case v of
    _ : Vector -> ...
    _ : Array -> This will _never_ be reached anymore
```

But for most purposes maybe the conversion could work indeed?

Reply:
will put back anyway (though I hate them still being about), but we should think on case statements and see if we can get rid of it.

We might then need Array before Vector if we want to have to branches.

Reply:
Fair. I'm hesitant of allowing in case, as then the programmer will need to be careful about branch order to avoid accidental conversions - this feels like pain to debug. But I agree we need to see what will work best here.